src/humanize.cr

```diff
@@ -23,7 +23,16 @@ struct Number
     if decimal_places
       number = number.round(decimal_places)
     end
-    string = number.abs.to_s
+
+    if number.is_a?(Float)
+      string = String.build do |io|
+        # Make sure to avoid scientific notation of default Float#to_s
+        Float::Printer.print(number.abs, io, point_range: ..)
+      end
```

Comment:
For `Float64::MAX` to print properly, this should really be using `"%.*f" % {decimal_places, number.abs}` when `decimal_places` is given, otherwise we have to define a suitable precision ourselves. This can probably be achieved by invoking `Grisu3.grisu3` directly, but is definitely easier with #10913.

Also if `number.is_a?(Float)` it is probably a good idea to return early on infinity and NaN here.

Reply:
The problem is figuring out the right precision for the integer component. So we might want to wait for #10913 with that if that makes it easier.

Reply:
`Number#format` should print _everything_ to the left of the decimal point, only `#humanize` trims to a number of significant figures. Or did I misunderstand something?

Reply:
Yes, that's how it should work. But if `decimal_places` is nil, we can't use `sprintf` that easily. We would have to calculate the number of significant decimal places depending on the size of the floating point value. So that already happens in the float printer implementation, but there it's overall significant places, regardless of the position of the decimal point. Essentially, the default value would be `decimal_places = (significant_places - integer_places).clamp(0..)`